src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -110,6 +113,9 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
             result = resultRepository.save(result);
             result.setSubmission(tmpSubmission);
 
+            // update StudentScore
+            studentScoreService.updateResult(result);
```

Comment:
I am not an expert with entity listeners. Why is this needed when you have the entity listener?

Reply:
This is a special case that needs to be done separately.